core/shared/src/main/scala/zio/prelude/Equal.scala

```diff
@@ -246,6 +246,14 @@ object Equal extends Lawful[Equal] {
   implicit def DeriveEqual[F[_]: DeriveEqual, A: Equal]: Equal[F[A]] =
     Derive[F, Equal].derive(Equal[A])
 
+  /**
+   * `Equal` instance for `Double` for imprecise equality with configurable tolerance.
+   */
+  def DoubleEqualWithEpsilon(epsilon: Double = 1.0d / 1024 * 1024 * 1024 * 1024): Equal[Double] = {
```

Comment:
I would move these to test files. There is value in having some tests for the corresponding instances since we decided to have them but this is kind of specific to testing them and these equality instances are not lawful in the sense that for example you could have `a` and `b` be equal within tolerance and `b` and `c` be equal within tolerance but not `a` and `c` being equal within tolerance.

Reply:
You're right that the instance created by `DoubleEqualWithEpsilon` wouldn't be lawful (unless `epsilon` = 0).
But these kinds of instances will still be practically useful. Not just for us testing various other Type classes, but even more importantly to our users. In practice, when you're working with floating point numbers -- geometry and alike come to mind, but this is a widespread phenomenon -- you very often don't want 100% precise comparison. You need to compare floats within a small "epsilon" tolerance.

That's why `DoubleEqualWithEpsilon` needs to stay in `main`, so that it's accessible to our users.

Reply:
That's a good point. I wonder if it makes sense to think of this as a separate abstraction, `TestEqual` or `ApproximatelyEqual`, but let's get this in and we can discuss that.